### ISSUE TITLE
aws_cloudfront_distribution resource creation panics on empty value in restriction_type

### ISSUE THREAD

**Issue by mrf** [terraform-providers/terraform-provider-aws#10668]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.12.12
+ provider.aws v2.33.0

### Affected Resource(s)
* aws_cloudfront_distribution

### Terraform Configuration Files

```hcl
resource "aws_s3_bucket" "my_bucket" {
  bucket = "testing-terraform-provider-my-bucket"
  acl = "private"
}

resource "aws_cloudfront_distribution" "s3_distribution" {
	origin {
		domain_name = "${aws_s3_bucket.my_bucket.id}.s3.amazonaws.com"
		origin_id = "myS3Origin"
	}
	enabled = true
	default_root_object = "index.html"
	default_cache_behavior {
		allowed_methods = [ "DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT" ]
		cached_methods = [ "GET", "HEAD" ]
		target_origin_id = "myS3Origin"
		forwarded_values {
			query_string = false
			cookies {
				forward = "none"
			}
		}
		viewer_protocol_policy = "allow-all"
		min_ttl = 0
		default_ttl = 3600
		max_ttl = 86400
	}
	price_class = "PriceClass_200"
	restrictions {
		geo_restriction {
			restriction_type = ""
		}
	}
	viewer_certificate {
		cloudfront_default_certificate = true
	}
	tags = {
            environment = "production"
            account = "main"
	}
}
```

### Debug Output
https://gist.github.com/mrf/acf495ff2ec424d404c56537f18a581f

### Panic Output
https://gist.github.com/mrf/03f3095bd71f658c0f3be2064deda693

### Expected Behavior
I should see an error message letting me know that restriction_type cannot be empty.

### Actual Behavior
```
aws_cloudfront_distribution.cdn-distribution: Creating...

Error: rpc error: code = Unavailable desc = transport is closing
```

### Steps to Reproduce
1. Edit restrictions block in resource to have empty value for restriction_type. Used test resource testAccAWSCloudFrontDistributionS3ConfigWithTags from acceptance tests in above example.
2. `terraform plan`
3. `terraform apply`

**Comment by kenXengineering** [comment]
As a workaround for now, you can set `restriction_type` to `none`.

**Comment by bflad** [comment]
This should have been fixed with https://github.com/terraform-providers/terraform-provider-aws/pull/12168, which was released a few months ago in version 2.52.0 of the Terraform AWS Provider. If you are still having trouble with this, please open a new issue and we will take a fresh look, thanks.
